retworkx/__init__.py

```diff
@@ -398,6 +398,74 @@ def _graph_dijkstra_shortest_path(
     )
 
 
+@functools.singledispatch
+def all_pairs_dijkstra_shortest_paths(graph, edge_cost_fn):
+    """Find the shortest path from all nodes
+
+    This function will generate the shortest path from all nodes int the graph
+    using Dijkstra's algorithm. This function is multithreaded and will run
+    launch a thread pool with threads equal to the number of CPUs by default.
+    You can tune the number of threads with the ``RAYON_NUM_THREADS``
+    environment variable. For example, setting ``RAYON_NUM_THREADS=4`` would
+    limit the thread pool to 4 threads.
+
+    :param graph: The input graph to use. Can either be a
+        :class:`~retworkx.PyGraph` or :class:`~retworkx.PyDiGraph`
+    :param edge_cost_fn: An optional weight function for an edge. It will
```

Comment:
Is `edge_cost_fn` really optional?

Reply:
Good catch, yeah it's required here it's optional in some of the other functions (where I copied the docstring from)

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/278/commits/e9ee3657eab1cc02bcfa7cb5829e32ce2a23d1e3